src/solver/conv_mlir_igemm_fwd_xdlops.cpp

```diff
@@ -139,14 +151,35 @@ bool PerformanceConvMlirIgemmXdlops::SetNextValue(const ConvolutionContext& /*co
             break;
         if(!NextTwoPower<16, 256>(GemmNPerBlock))
             break;
-        if(!NextTwoPower<1, 8>(GemmKPerBlock))
+        if(!NextTwoPower<4, 8>(GemmKPerBlock))
             break;
         if(!NextTwoPower<4, 128>(GemmMPerWave))
             break;
         if(!NextTwoPower<16, 128>(GemmNPerWave))
             break;
-        if(!NextTwoPower<4, 8>(GemmKPACKSize))
-            break;
+        if(config.IsFp32())
+        {
+            if(config.direction.IsForward())
+            {
+                if(!NextTwoPower<1, 4>(GemmKPACKSize))
+                    break;
+            }
+            else if(config.direction.IsBackwardWrW())
+            {
+                if(!NextTwoPower<1, 2>(GemmKPACKSize))
+                    break;
+            }
+            else if(config.direction.IsBackwardData())
+            {
+                if(!NextTwoPower<1, 1>(GemmKPACKSize))
```

Comment:
I thought you have started another PR to enable kpack for bwd, if so, this condition is out-of-date?

Reply:
This PR does not aim to support enable KPACK for bwd yet, therefore KPACK is fixed as 1 for now.

Another PR will be filed once the MLIR PR gets merged.

Reply:
Gotcha, sounds good.